TACC/Core-Styles
# Task/fp 1878 table pattern

## Overview

Style basic tables, tables with paragraphs, and nested tables.

## Related

- [FP-1878](https://jira.tacc.utexas.edu/browse/FP-1878)

## Changes

- add a one paragraph text snippet
- change `<li>` line-height to not apply to tables
- add table styles: basic, paragraph, nested

## Testing

1. Open [/components/detail/table--basic](http://localhost:3000/components/detail/table--basic).
    1. Verify styles match [design][b design].
    2. On narrow frame[^1], verify:
        1. first column is pinned
        2. other columns scroll
        3. no style oddities
2. Open [/components/detail/table--paragraph](http://localhost:3000/components/detail/table--paragraph).
    1. Verify styles match [design][p design] _e.g. cell content is top-aligned_.
    2. On narrow frame[^1], verify:
        1. first column is pinned
        2. other columns scroll
        3. no style oddities
3. Open [/components/detail/table--nested](http://localhost:3000/components/detail/table--nested).
    1. Verify styles match [design][n design].
    2. On narrow frame[^1], verify:
        1. first column is pinned
        2. other columns scroll
        3. no style oddities
    2. On **very** narrow frame[^2], verify:
        1. first column, of nested table, is _also_ pinned
        2. other columns, of nested table, scroll
        3. no style oddities

[^1]: Narrow enough that table content can not horizontally fit.
[^2]: Narrow enough that nested table content can not horizontally fit.

## UI

### Basic

| Wide | Narrow |
| - | - |
| <img width="1312" alt="basic wide" src="https://user-images.githubusercontent.com/62723358/201543639-650b5b98-74cd-4f41-8397-9c38dde3263f.png"> | <img width="1312" alt="basic narrow" src="https://user-images.githubusercontent.com/62723358/201543638-79e6672c-bb2b-4080-a72f-85d30b8cec8d.png"> |

### Paragraph

| Wide | Narrow | Very Narrow |
| - | - | - |
| <img width="1312" alt="paragraph wide" src="https://user-images.githubusercontent.com/62723358/201543637-f801be3f-10ee-4952-a335-c7475db10aa6.png"> | <img width="1312" alt="paragraph narrow" src="https://user-images.githubusercontent.com/62723358/201543634-90febb07-61bd-4b7c-af4c-adab2483bce4.png"> | <img width="1312" alt="paragraph very narrow" src="https://user-images.githubusercontent.com/62723358/201543632-3aa58674-b5a6-4eef-a9c2-4227962f947a.png"> |

### Nested

| Wide | Narrow | Very Narrow |
| - | - | - |
| <img width="1312" alt="nested wide" src="https://user-images.githubusercontent.com/62723358/201543636-b3fe04cf-c75a-4ab8-a538-e639e607e076.png"> | <img width="1312" alt="nested narrow" src="https://user-images.githubusercontent.com/62723358/201543633-2e9a4072-6501-43c3-b1b2-473864bb6a6f.png"> | <img width="1312" alt="nested very narrow" src="https://user-images.githubusercontent.com/62723358/201543631-e30e55f0-05d2-42db-84da-9698ec7170d5.png"> |

## Notes

- [basic table design][b design]
- [paragraph table design][p design]
- [nested table design][n design]

[b design]: https://xd.adobe.com/view/b37b291a-7798-4814-90f3-7ac63c48a3e0-e628/screen/907a5f5c-5b46-441f-95f6-f0e50f529c65/specs/ "Basic Table Design"
[p design]: https://xd.adobe.com/view/b37b291a-7798-4814-90f3-7ac63c48a3e0-e628/screen/7ecf7f47-e43b-4839-a8b4-a4b51777fa5e/ "Paragraph Table Design"
[n design]: https://xd.adobe.com/view/db2660cc-1011-4f26-5d31-019ce87c1fe8-ad17/screen/88c12116-9fe4-429a-ae4d-d66eb1fa2a61/specs/ "Nested Table Design"
